NitroxServer/Serialization/World/WorldPersistence.cs

```diff
@@ -26,14 +26,16 @@ public class WorldPersistence
         public IServerSerializer Serializer { get; private set; }
         private string FileEnding => Serializer?.FileEnding ?? "";
 
+        private readonly IWorldModifier worldModifier;
         private readonly ServerProtoBufSerializer protoBufSerializer;
         private readonly ServerJsonSerializer jsonSerializer;
         private readonly ServerConfig config;
         private readonly RandomStartGenerator randomStart;
         private readonly SaveDataUpgrade[] upgrades;
 
-        public WorldPersistence(ServerProtoBufSerializer protoBufSerializer, ServerJsonSerializer jsonSerializer, ServerConfig config, RandomStartGenerator randomStart, SaveDataUpgrade[] upgrades)
+        public WorldPersistence(IWorldModifier worldModifier, ServerProtoBufSerializer protoBufSerializer, ServerJsonSerializer jsonSerializer, ServerConfig config, RandomStartGenerator randomStart, SaveDataUpgrade[] upgrades)
```

Comment:
Very minor/preference: Would pack the modifier (declaration and parameter) before or after the upgrades